### ISSUE TITLE
Error on https://ooooooooo.ooo/static/

### ISSUE THREAD

**Issue by Greensky-gs** [ruffle-rs/ruffle#13048]
# Error Info
Error name: Error
Error message: panicked at 'Unexpected non-object value in object vector', core/src/avm2/amf.rs:99:34
Error stack:
```
Error: panicked at 'Unexpected non-object value in object vector', core/src/avm2/amf.rs:99:34
    at V.n.wbg.__wbg_new_d258248ed531ff54 (https://unpkg.com/@ruffle-rs/ruffle/core.ruffle.9cdaa3ec5f23e9e60beb.js:1:51650)
    at core::ops::function::FnOnce::call_once{{vtable.shim}}::h3959f346f7109267 (wasm://wasm/0306993a:wasm-function[9258]:0x9138dc)
    at std::panicking::rust_panic_with_hook::h7601402c0a383194 (wasm://wasm/0306993a:wasm-function[6015]:0x8139ef)
    at core::panicking::panic_fmt::h6dad0405f48e39e2 (wasm://wasm/0306993a:wasm-function[9098]:0x90efb5)
    at core::option::expect_failed::h980aad86598061bd (wasm://wasm/0306993a:wasm-function[7989]:0x8cd26c)
    at ruffle_core::avm2::amf::serialize_value::h0c6e51a27b6adf32 (wasm://wasm/0306993a:wasm-function[757]:0x264ea4)
    at ruffle_core::avm2::amf::recursive_serialize::hc449dd3981488c2f (wasm://wasm/0306993a:wasm-function[2399]:0x5ae9a6)
    at ruffle_core::avm2::globals::flash::net::shared_object::new_lso::h202a3cd4a0ee1271 (wasm://wasm/0306993a:wasm-function[3004]:0x64d064)
    at ruffle_core::avm2::globals::flash::net::shared_object::flush::ha1f8433d7a91916a (wasm://wasm/0306993a:wasm-function[1471]:0x44c0e8)
    at ruffle_core::avm2::function::Executable::exec::hebfea0b7e1b2fb7d (wasm://wasm/0306993a:wasm-function[829]:0x2b778d)
```
AVM2 stack:
```
    at flash.net::SharedObject/flash::net::SharedObject::flush()
    at com.spilgames.api.core.tracking.ext.spilet.trackStates::NotSpilState/updateSharedObject()
    at com.spilgames.api.core.tracking.ext.spilet.trackStates::NotSpilState/addElementToQueue()
    at com.spilgames.api.core.tracking.ext.spilet.trackStates::NotSpilState/startSession()
    at com.spilgames.api.core.tracking.ext.spilet.trackStates::NotSpilState/init()
    at com.spilgames.api.core.tracking.ext.spilet.trackStates::NotSpilState()
    at com.spilgames.api.core.tracking.ext.spilet.setTracker::SET$/setEnvironment()
    at com.spilgames.api.core.tracking.ext.spilet.setTracker::SET$/init()
    at com.spilgames.api.core.tracking.ext.spilet.setTracker::SET$/getInstance()
    at com.spilgames.api.core.tracking.ext.spilet::SETTracker/setup()
    at com.spilgames.api.core.tracking.ext::ATracker/setClass()
    at com.spilgames.api.core.tracking.ext::TrackerLoader/load()
    at com.spilgames.api.core.tracking::TrackController/loadLibraries()
    at com.spilgames.api.core.tracking::TrackController/onLoadConfig()
    at com.spilgames.api.core.tracking::TrackConfig/onLoadConfig()
```

# Player Info
Allows script access: true
Renderer: wgpu
Adapter Backend: Gl
Adapter Name: "ANGLE (NVIDIA, NVIDIA GeForce GT 730 Direct3D11 vs_5_0 ps_5_0, D3D11)"
Adapter Device Type: Other
Adapter Driver Name: ""
Adapter Driver Info: ""
Enabled features: Features(TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES | PUSH_CONSTANTS)
Available features: Features(TEXTURE_COMPRESSION_BC | CLEAR_TEXTURE | MULTIVIEW)
Current limits: Limits { max_texture_dimension_1d: 16384, max_texture_dimension_2d: 16384, max_texture_dimension_3d: 2048, max_texture_array_layers: 256, max_bind_groups: 4, max_bindings_per_bind_group: 640, max_dynamic_uniform_buffers_per_pipeline_layout: 8, max_dynamic_storage_buffers_per_pipeline_layout: 0, max_sampled_textures_per_shader_stage: 16, max_samplers_per_shader_stage: 16, max_storage_buffers_per_shader_stage: 0, max_storage_textures_per_shader_stage: 0, max_uniform_buffers_per_shader_stage: 11, max_uniform_buffer_binding_size: 16384, max_storage_buffer_binding_size: 0, max_vertex_buffers: 8, max_buffer_size: 268435456, max_vertex_attributes: 16, max_vertex_buffer_array_stride: 255, min_uniform_buffer_offset_alignment: 256, min_storage_buffer_offset_alignment: 256, max_inter_stage_shader_components: 60, max_compute_workgroup_storage_size: 0, max_compute_invocations_per_workgroup: 0, max_compute_workgroup_size_x: 0, max_compute_workgroup_size_y: 0, max_compute_workgroup_size_z: 0, max_compute_workgroups_per_dimension: 0, max_push_constant_size: 96 }
Surface quality: high
Surface samples: 4
Surface size: Extent3d { width: 550, height: 400, depth_or_array_layers: 1 }

# Page Info
Page URL: https://ooooooooo.ooo/static/?e4957df5-9f4c-7e0e-ed60-3de2ed79023c
SWF URL: http://cache.armorgames.com/files/games/pocket-ninja-15843.swf

# Browser Info
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36
Platform: Win32
Has touch support: false

# Ruffle Info
Version: 0.1.0
Name: nightly 2023-09-03
Channel: nightly
Built: 2023-09-03T00:19:43.518Z
Commit: 2996f97578f4b9541029f0318d7b87a60664cf23
Is extension: false

# Metadata
width: 640
height: 480
frameRate: 30
numFrames: 3
swfVersion: 12
backgroundColor: #666666
isActionScript3: true
uncompressedLength: 6615873

**Comment by n0samu** [comment]
Same problem as #12366

**Comment by n0samu** [comment]
Fixed by #11971
